File: Array/15. Mean And Median of Array - GFG/15.-mean-and-median-of-array.cpp
// { Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;


 // } Driver Code Ends
//User function Template for C++

class Solution{
    public:
    //Function to find median of the array elements.
    int median(int A[],int N)
    {
        
        sort(A,A+N);
        if ( N % 2 != 0){
            return A[N/2];
        }
        else {
            return int((A[(N-1)/2] + A[(N-1)/2 + 1]) / 2);
        }
        
        //your code here
        //If median is in fraction then convert it to integer and return
    }
    //Function to find mean of the array elements.
    int mean(int A[],int N)
    {
        //your code here
        int sum = 0;
        for ( int i = 0; i < N; i++){
            sum = sum + A[i];
        }
        return sum/N;
    }
};


// { Driver Code Starts.

int main()
{
    //testcase
    int T;
    cin>>T;
   
    //looping testcase
    while(T--)
    {
        //number of elements in array
        int N;
        cin>>N;
        int a[N];
        
        //inseting elements in the array
        for(int i=0;i<N;i++){
            cin>>a[i];
        }
        Solution ob;
        //calling mean() and median() functions
        cout<<ob.mean(a,N)<<" "<<ob.median(a,N)<<endl;
    }
    return 0;
}
  // } Driver Code Ends

File: 15. Mean And Median of Array - GFG/15.-mean-and-median-of-array.py
#{ 
#Driver Code Starts
#Initial Template for Python 3


import math



 # } Driver Code Ends
#User function Template for python3

class Solution:
    ##Complete the below codes
    #Function to find median of the array elements.
    def median(self,A,N):
        
        A.sort()
        
        ##Your code here
        #If median is fraction then convert the median to integer and return
        if N % 2 == 0:
            return (A[(N-1)//2] + A[(N-1)//2 + 1]) // 2
        else:
            return A[N//2]
     
    #Function to find mean of the array elements.   
    def mean(self,A,N):
        ##Your code here
        return sum(A)//N

#{ 
#Driver Code Starts.
    
def main():
    
    T=int(input())
    
    while(T>0):
        
        N=int(input())
        a=[int(x) for x in input().strip().split()]
        ob=Solution()
        print(ob.mean(a,N),ob.median(a,N))
        
        T-=1
    
    




if __name__=="__main__":
    main()
#} Driver Code Ends